source/vstd/set.rs

```diff
@@ -291,12 +310,28 @@ pub broadcast proof fn axiom_set_empty_finite<A>()
     admit();
 }
 
+/// The result of inserting an element `a` into a set `s` is finite iff `s` is finite.
+pub broadcast proof fn axiom_set_insert_finite_iff<A>(s: Set<A>, a: A)
+    ensures
+        #[trigger] s.insert(a).finite() <==> s.finite(),
+{
+    admit();
+}
+
```

Comment:
It seems like this isn't really necessary:

```
    pub broadcast proof fn axiom_set_insert_finite_iff<A>(s: Set<A>, a: A)
        ensures
            #[trigger] s.insert(a).finite() <==> s.finite()
    {
        if s.insert(a).finite() {
            if s.contains(a) {
                assert(s == s.insert(a));
            } else {
                assert(s == s.insert(a).remove(a));
            }
        }
    }
```

Reply:
Well now I feel like a clown :)

Thanks for pointing that out. I totally missed that the axiom for `remove` effectively functions as the inverse axiom I was looking for.

I now wrote similar proofs for `remove` and `union` and moved everything to `set_lib.rs` instead. I added you as a co-author on the commit. Hope that's okay.